### PR TITLE
feat: added config for creating fields based on form configuration

### PR DESCRIPTION
**Pull request by sksadjad**
This is based on one of our previous environments. env specific changes will be in another pr

### CODE REVIEW

File: packages/oid4vci-demo-frontend/public/locales/development/en/translation.json [hunk 1]
```diff
@@ -36,5 +36,12 @@
   "sharing_data_success_right_pane_paragraph_short": "for providing the information. We are now ready\r\n to issue a new credential to you.",
   "sharing_data_success_right_pane_title": "Thank you {{firstName}}!",
   "sharing_data_success_get_mobile_app_message": "Get the Sphereon Wallet",
-  "action_open_wallet_caption": "Open in wallet"
+  "action_open_wallet_caption": "Open in wallet",
+  "ssi_information_request_page_form_company_name_title": "Company name",
+  "ssi_information_request_page_form_coc_number_title": "KVK number",
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/public/locales/development/en/translation.json]: CoC

File: packages/oid4vci-demo-frontend/src/ecosystem-config.ts [hunk 2]
```diff
@@ -38,12 +41,15 @@ export function getCurrentEcosystemConfig(): VCIConfig {
...
         default:
             return sphereon as VCIConfig;
     }
 }
 
 export function getCurrentEcosystemPageOrComponentConfig(pageOrComponent: string): PageOrComponentConfig {
+    console.log(getCurrentEcosystem())
     switch (getCurrentEcosystem()) {
         case VCIEcosystem.fmdm:
             return getEcosystemPageOrComponentConfig(pageOrComponent, fmdm as VCIConfig);
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: Let's make a ticket, because we are simply getting the enum and exteding the switch statements. Then we are basically using the enum value as argument everywhere. We need to come up with a plan to resolve names/configs automatically and just have an env var determine what we will serve by default

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    return data.flat().map(field => field.key);
+};
+
+export const extractRequiredKeys = (data: DataFormRow[]): string[] => {
+  return data.flat().filter(field=> field.required).map(field => field.key);
+};
+
+export const transformDataToObject = (data: DataFormRow[]): Record<string, string> => {
+    return data.flat().reduce((acc, field) => {
+        acc[field.key] = '';
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: Why? Also I would expect this to use a flatMap.

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx [hunk 4]
```diff
@@ -12,38 +12,59 @@ import {
...
+function getInitialState(form: DataFormRow[] | undefined) {
+  if (!form) {
+    return {
+      firstName: '',
+      lastName: '',
+      emailAddress: ''
+    }
+  }
+  return form.flat().reduce((acc, field) => {
+    acc[field.key] = '';
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx]: Why?

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import {DataFormRow} from "../ecosystem-config";
+
+export const extractKeys = (data: DataFormRow[]): string[] => {
+    return data.flat().map(field => field.key);
+};
+
+export const extractRequiredKeys = (data: DataFormRow[]): string[] => {
+  return data.flat().filter(field=> field.required).map(field => field.key);
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: Why not use a flatmap to begin with?

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import {DataFormRow} from "../ecosystem-config";
+
+export const extractKeys = (data: DataFormRow[]): string[] => {
+    return data.flat().map(field => field.key);
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: Why not use a flatmap to begin with

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks.

File: packages/oid4vci-demo-frontend/src/ecosystem-config.ts (hunk 2 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: I've partly addressed this in current pr (removed enum value and reduced complexity of it) but for the rest created another [ticket](https://sphereon.atlassian.net/browse/VDX-251)

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts (hunk 3 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: replaced with flatmap. and for your question, I think you’re talking about assigning the empty string to the keys, we have been doing that for other form inputs such as firstName, lastName, and email (https://github.com/Sphereon-Opensource/OID4VC-demo/blame/16bcb3df57fdf1b804b25cc9f2dd396fbd9e5863/packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx#L42).

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx (hunk 4 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx]: changed this part of the logic to get the empty object from the ObjectUtils method and then populate it here from the credentialSubject

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts (hunk 5 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: changed to flatmap

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts (hunk 6 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: changed to flatmap

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts (hunk 3 above)
Comment (nklomp) [reply packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: Okay then suggest to rename this method, because that name doesn't suggest it

File: packages/oid4vci-demo-frontend/src/ecosystem-config.ts [hunk 7]
```diff
@@ -1,61 +1,41 @@
...
-        case VCIEcosystem.dbc:
-            return dbc as VCIConfig;
-        case VCIEcosystem.energy_shr:
-            return energy_shr as VCIConfig;
-        default:
-            return sphereon as VCIConfig;
-    }
+  const ecosystem = process.env.REACT_APP_ENVIRONMENT ?? 'sphereon';
+  switch (ecosystem) {
+    case 'dbc':
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: Just do ecosystem as VCIConfig

File: packages/oid4vci-demo-frontend/public/locales/development/en/translation.json (hunk 1 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/public/locales/development/en/translation.json]: changed

File: packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts (hunk 3 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts]: renamed it to `transformFormConfigToEmptyObject`

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx [hunk 8]
```diff
@@ -80,16 +98,26 @@ const SSIInformationRequestPage: React.FC = () => {
...
+                return {
+                    firstName: cs.firstName,
+                    lastName: cs.lastName,
+                    emailAddress: cs.emailAddress
+                } as Record<string, string>;
+            }
+            const payload = transformFormConfigToEmptyObject(form);
+            for (const payloadKey in payload) {
+                if (payloadKey in cs) {
+                    // TODO: since this code is based on the manual flow, we have to revisit it for the wallet flow
```
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx:116]: What do you mean with this token. This is the wallet flow right?

**Review by nklomp — CHANGES_REQUESTED** [review]

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx (hunk 8 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx:116]: yeah, since I wrote the code for manual flow, just changed a bit to match the wallet flow as well, but it needs revisiting when we're going to use it for wallet flow

File: packages/oid4vci-demo-frontend/src/ecosystem-config.ts (hunk 7 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: I'm not sure what do you mean by this. ecosystem string is the name of the ecosystem. and for importing dynamically we need it to be asynchronous. and I've created another ticket for that

Comment (nklomp) [reply packages/oid4vci-demo-frontend/src/ecosystem-config.ts:21]: You are litterlay doing the same every time based on the key (ecosystem), Just do `ecosystem as VCIConfig` and be done with it. The only change in behaviour would be that there is no fallback to Sphereon anymore, which there shouldn't be anyway

Comment (nklomp) [reply packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: You do not need async. You could just use the filesystem instead of importing it at the top

Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/ecosystem-config.ts]: changed the logic here

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx (hunk 8 above)
Comment (nklomp) [reply packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx:116]: But only the first issuer has a manual flow, the other issuers use a wallet flow. So I guess this needs more work

**Review by nklomp — CHANGES_REQUESTED** [review]
Given the logic for the manual flows. Only the first issuer has a manual flow, the other issuers use a wallet flow. So I guess this PR needs more work

File: packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx (hunk 8 above)
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx:116]: tested it for the wallet flow as well. it still needs more work regarding credential issuance. but for this ticket, it's done

File: packages/oid4vci-demo-frontend/src/configs/notary.json
Comment (nklomp) [inline packages/oid4vci-demo-frontend/src/configs/notary.json:1]: Is this file really needed? I am a bit worried about work being done by others when merging this. It does not contain notary info, so I rather would have seen that we have updated an existing config to support any new keys. Or better yet document that a bit. It now is hard to determine what in this file is new compared to what we had.

**Review by nklomp — CHANGES_REQUESTED** [review]
See remark about the config file. Rest looks good

File: packages/oid4vci-demo-frontend/src/configs/notary.json
Comment (sksadjad) [reply packages/oid4vci-demo-frontend/src/configs/notary.json:1]: this file is a base, I've changed it in my other pr, for WAL-641

**Review by nklomp — APPROVED** [review]
